processor/attributekeyprocessor/testdata/config.yaml

```diff
@@ -0,0 +1,30 @@
+receivers:
+  examplereceiver:
+
+processors:
+  attribute-key:
+    key-replacements:
```

Comment:
The old config format is not very user friendly. We need to change the configuration format to use a mapping instead of array. So the config should look like this:
```
key-replacements:
  foo:
    replacement: "boo"
    overwrite: true
    keep: true
  bar:
    replacement: "another string"
    overwrite: false
    keep: true
```
This is also uniform with how other mappings are specified (e.g. receivers, processors, etc). Sorry for not specifying the details upfront in the issue description.